### PR TITLE
Kill some of the old sharing code related to share backends

### PR DESCRIPTION
**Pull request by PVince81**
## Description
Remove the "File" and "Folder" item type and share backends from files_sharing.
These should not be needed any more when using the OCS Share API.

Also removes any reference to file/folder in the old share API in `\OC\Share` (where possible), so that the old sharing stays only for other types like contacts/etc. (another approach would be to rewire these API to the new API if a file/folder item type is given ?)

Also remove most methods in \OC:Share class

## Related Issue
General cleanup issue: https://github.com/owncloud/core/issues/22209
Issue about those backends using oc_filecache: https://github.com/owncloud/core/issues/26607

## Motivation and Context
Removes the old code,

## How Has This Been Tested?
- Quick test in the web UI, sharing still seems to work, so does "unshare from self".
- A quick run of integration tests locally showed that many calls still work (but my env is a bit broken so want to see that run on CI).
- Will need manual testing of all share scenarios (and/or complete these as integration tests)
- Retest fed sharing

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## TODOs:

- [ ] port all file share related tests that use `shareItem` with file/folder to use the share manager instead

@DeepDiver1975 @jvillafanez

### CODE REVIEW

**Comment by PVince81** [comment]
Looks like `View::isTargetAllowed` is a big blocker for many tests.
Might want to move this to the `SharedMount` implementation and add that method on the `MoveableMount` interface.

But the bigger problem is that it seems that the share manager has no replacement for `\OCP\Share::getItemShared`

**Comment by PVince81** [comment]
Solved a few test issues by porting the tests to the new API.
Unfortunately most are failing due to https://github.com/owncloud/core/pull/26608#issuecomment-259991602. Need a good solution.

**Comment by PVince81** [comment]
- [ ] port publicpreview.php to share 2.0: https://github.com/owncloud/core/pull/26611 (also included in this PR)

**Comment by PVince81** [comment]
Remaining to port:
```
apps/files_sharing/tests/DeleteOrphanedSharesJobTest.php:               \OCP\Share::registerBackend('test', 'Test\Share\Backend');
apps/files_sharing/lib/ExpireSharesJob.php:                     \OCP\Share::unshare($share['item_type'], $share['file_source'], \OCP\Share::SHARE_TYPE_LINK, null, $share['uid_owner']);
apps/files_sharing/lib/Helper.php:              $linkItem = \OCP\Share::getShareByToken($token, !$password);
apps/files_sharing/lib/Helper.php:              $rootLinkItem = \OCP\Share::resolveReShare($linkItem);
lib/private/Encryption/File.php:                        $resultForParents = \OCP\Share::getUsersSharingFile($parent, $owner);
lib/private/Encryption/File.php:                $resultForFile = \OCP\Share::getUsersSharingFile($ownerPath, $owner, false, false, false);
lib/private/Files/View.php:             $shares = \OCP\Share::getItemShared(
lib/private/Share/Share.php:            if ($checkPasswordProtection && !\OCP\Share::checkPasswordProtectedShare($row)) {
```

**Comment by PVince81** [comment]
- [ ] getting rid of Maintainer.php: https://github.com/owncloud/core/pull/26612

**Comment by PVince81** [comment]
This should help getting rid of `getItemShared` => https://github.com/owncloud/core/pull/26615

**Comment by PVince81** [comment]
- [ ] kill `\OCP\Share` altogether as @DeepDiver1975 told me that it's not used by calendar and contacts any more :tada:

**Comment by PVince81** [comment]
Rebased as the `isTargetAllowed` fix was merged. Now there should be much less test failures.

**Comment by PVince81** [comment]
```
18:44:53 1) OCA\Files_Sharing\Tests\ApiTest::testDefaultExpireDate
18:44:53 Exception: Sharing backend for file not found
18:44:53 
18:44:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/lib/private/Share/Share.php:1474
18:44:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/lib/private/Share/Share.php:603
18:44:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/lib/public/Share.php:267
18:44:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/apps/files_sharing/tests/ApiTest.php:1399
18:44:53 
18:44:53 2) OCA\Files_Sharing\Tests\ExpireSharesJobTest::testExpireLinkShare with data set #2 (true, 'P1D', false, true)
18:44:53 Exception: Sharing backend for folder not found
18:44:53 
18:44:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/lib/private/Share/Share.php:1474
18:44:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/lib/private/Share/Share.php:863
18:44:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/lib/public/Share.php:281
18:44:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/apps/files_sharing/lib/ExpireSharesJob.php:71
18:44:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/apps/files_sharing/tests/ExpireSharesJobTest.php:175
18:44:53 
18:44:53 3) OCA\Files_Sharing\Tests\ExpireSharesJobTest::testExpireLinkShare with data set #4 (true, 'P1W', false, true)
18:44:53 Exception: Sharing backend for folder not found
18:44:53 
18:44:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/lib/private/Share/Share.php:1474
18:44:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/lib/private/Share/Share.php:863
18:44:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/lib/public/Share.php:281
18:44:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/apps/files_sharing/lib/ExpireSharesJob.php:71
18:44:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/apps/files_sharing/tests/ExpireSharesJobTest.php:175
18:44:53 
18:44:53 4) OCA\Files_Sharing\Tests\ExpireSharesJobTest::testExpireLinkShare with data set #6 (true, 'P1M', false, true)
18:44:53 Exception: Sharing backend for folder not found
18:44:53 
18:44:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/lib/private/Share/Share.php:1474
18:44:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/lib/private/Share/Share.php:863
18:44:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/lib/public/Share.php:281
18:44:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/apps/files_sharing/lib/ExpireSharesJob.php:71
18:44:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/apps/files_sharing/tests/ExpireSharesJobTest.php:175
18:44:53 
18:44:53 5) OCA\Files_Sharing\Tests\ExpireSharesJobTest::testExpireLinkShare with data set #8 (true, 'P1Y', false, true)
18:44:53 Exception: Sharing backend for folder not found
18:44:53 
18:44:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/lib/private/Share/Share.php:1474
18:44:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/lib/private/Share/Share.php:863
18:44:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/lib/public/Share.php:281
18:44:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/apps/files_sharing/lib/ExpireSharesJob.php:71
18:44:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/apps/files_sharing/tests/ExpireSharesJobTest.php:175
```

**Comment by PVince81** [comment]
apps/files_sharing/tests/ApiTest.php will need more work as it seems to test older APIs

**Comment by PVince81** [comment]
Note: we can't kill OCP\Share without killing ajax/share.php yet, so I'd do that in a separate PR.
And we can't kill ajax/share.php as the gallery app is still using some of these old endpoints.

**Comment by PVince81** [comment]
Last failure:
```
10:06:59 1) OCA\Files_Sharing\Tests\ApiTest::testDefaultExpireDate
10:06:59 Exception: Sharing backend for file not found
10:06:59 
10:06:59 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/lib/private/Share/Share.php:1474
10:06:59 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/lib/private/Share/Share.php:603
10:06:59 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/lib/public/Share.php:267
10:06:59 /var/lib/jenkins/workspace/owncloud-core_core_PR-26608-7HTQSAV7ASDXHK3HEP454Z4KEUYQXFTEM4XUGRAL27I4X6LVM6NQ/apps/files_sharing/tests/ApiTest.php:1399
```

Some old tests can be deleted but need to check that the new code has these tests too for the new sharing API.

**Comment by PVince81** [comment]
It is also likely that merging this PR would break gallery sharing, so need to verify that it doesn't.

File: apps/files_sharing/lib/ExpireSharesJob.php [hunk 1]
```diff
@@ -43,15 +64,12 @@ public function __construct() {
 	 */
 	public function run($argument) {
 		$connection = \OC::$server->getDatabaseConnection();
```
Comment (jvillafanez) [inline apps/files_sharing/lib/ExpireSharesJob.php]: I guess this should be replaced by `$this->connection`

File: apps/files_sharing/lib/ExpireSharesJob.php [hunk 2]
```diff
@@ -67,8 +85,13 @@ public function run($argument) {
...
 
 		$shares = $qb->execute();
-		while($share = $shares->fetch()) {
-			\OCP\Share::unshare($share['item_type'], $share['file_source'], \OCP\Share::SHARE_TYPE_LINK, null, $share['uid_owner']);
+		while ($share = $shares->fetch()) {
+			try {
+				// the getShareById already deletes those automatically
+				$this->shareManager->getShareById('ocinternal:' . $share['id']);
+			} catch (ShareNotFound $e) {
+				// ignore, already deleted
```
Comment (jvillafanez) [inline apps/files_sharing/lib/ExpireSharesJob.php]: I'd rather to log something for debugging purposes

**Comment by PVince81** [comment]
Gallery app sharing is broken, can't share with users. 😢

**Comment by PVince81** [comment]
@jvillafanez I fixed ExpireShareJobs based on your comments.

Remaining todos to move forward:
- [ ] find a solution with gallery
- [ ] remove most of the code from lib/private/Share/Share.php and other related stuff

**Comment by PVince81** [comment]
More work required before we can remove most/all methods from the old `Share` class:
```
apps/files_sharing/tests/DeleteOrphanedSharesJobTest.php|160| \OCP\Share::registerBackend('test', 'Test\Share\Backend');
apps/files_sharing/tests/DeleteOrphanedSharesJobTest.php|163| \OCP\Share::shareItem('test', 'test.txt', \OCP\Share::SHARE_TYPE_USER, $this->user2, \OCP\Constants::PERMISSION_READ),
apps/files_sharing/lib/Controller/ShareesController.php|487| $backend = Share::getBackend($itemType);
apps/files_sharing/lib/External/Storage.php|350| if (\OCP\Util::isSharingDisabledForUser() || !\OC\Share\Share::isResharingAllowed()) {
apps/files_sharing/lib/Helper.php|56| $linkItem = \OCP\Share::getShareByToken($token, !$password);
apps/files_sharing/lib/Helper.php|69| $rootLinkItem = \OCP\Share::resolveReShare($linkItem);
apps/files_sharing/lib/SharedStorage.php|193| if (\OCP\Util::isSharingDisabledForUser() || !\OC\Share\Share::isResharingAllowed()) {
apps/comments/lib/Activity/Listener.php|98| $users = array_merge($users, Share::getUsersSharingFile($path, $owner, true, true));
apps/federatedfilesharing/tests/RequestHandlerTest.php|75| \OCP\Share::registerBackend('test', 'Test\Share\Backend');
apps/systemtags/lib/Activity/Listener.php|171| $users = array_merge($users, Share::getUsersSharingFile($path, $owner, true, true));
lib/private/Encryption/File.php|74| $resultForParents = \OCP\Share::getUsersSharingFile($parent, $owner);
lib/private/Encryption/File.php|82| $resultForFile = \OCP\Share::getUsersSharingFile($ownerPath, $owner, false, false, false);
lib/private/Files/FileInfo.php|210| if (\OCP\Util::isSharingDisabledForUser() || ($this->isShared() && !\OC\Share\Share::isResharingAllowed())) {
lib/private/Share/Helper.php|49| $backend = \OC\Share\Share::getBackend($itemType);
lib/private/Share/Helper.php|178| $expireAfter = \OC\Share\Share::getExpireInterval() * 24 * 60 * 60;
lib/private/Share/Helper.php|303| $normalizedServer1 = strtolower(\OC\Share\Share::removeProtocolFromUrl($server1));
lib/private/Share/Helper.php|304| $normalizedServer2 = strtolower(\OC\Share\Share::removeProtocolFromUrl($server2));
lib/private/Share/MailNotifications.php|234| return Share::getItemSharedWithUser($itemType, $itemSource, $recipient->getUID());
lib/private/Share/Share.php|502| if ($checkPasswordProtection && !\OCP\Share::checkPasswordProtectedShare($row)) {
lib/private/Tags.php|129| $this->owners = array_merge($this->owners, \OC\Share\Share::getSharedItemsOwners($this->user, $this->type, true));
lib/private/Tags.php|130| $this->backend = \OC\Share\Share::getBackend($this->type);
lib/private/legacy/template.php|152| if (\OCP\Share::isEnabled()) {
core/js/config.php|168| 'resharingAllowed' => \OCP\Share::isResharingAllowed(),
core/ajax/share.php|78| \OCP\Share::setSendMailStatus($itemType, $itemSource, $shareType, $recipient, true);
core/ajax/share.php|97| \OCP\Share::setSendMailStatus($itemType, $itemSource, $shareType, $recipient, false);
core/ajax/share.php|133| $share = \OCP\Share::getShareByToken($token, false);
core/ajax/share.php|177| $return = OCP\Share::getItemsShared((string)$_GET['itemType'], OCP\Share::FORMAT_STATUSES);
core/ajax/share.php|187| $reshare = OCP\Share::getItemSharedWithBySource(
core/ajax/share.php|198| $shares = OCP\Share::getItemShared(
core/ajax/share.php|245| $shareWithinGroupOnly = OC\Share\Share::shareWithGroupMembersOnly();
core/ajax/share.php|331| $backend = \OCP\Share::getBackend((string)$_GET['itemType']);
tests/lib/Share/Backend.php|50| $shares = \OCP\Share::getItemsSharedWithUser('test', $shareWith);
tests/lib/TagsTest.php|289| \OCP\Share::registerBackend('test', 'Test\Share\Backend');
tests/lib/TagsTest.php|308| \OCP\Share::shareItem('test', 1, \OCP\Share::SHARE_TYPE_USER, $otherUserId, \OCP\Constants::PERMISSION_READ);
```

**Comment by PVince81** [comment]
Another idea to be able to kill gallery's old sharing code: https://github.com/owncloud/gallery/issues/727

This is easier and quicker to implement than https://github.com/owncloud/core/issues/17357. The fact it would use the files app sidebar means we can get rid of the gallery's share dialog code as it would simply be relying on the sidebar's sharing tab for that.

**Comment by PVince81** [comment]
Ok, so gallery will need to be reworked anyway for 10.1 considering the changes to the preview API, etc.

**Comment by PVince81** [comment]
Remaining `OC\Share::` methods that need further work for deletion:
- Share::checkPasswordProtectedShare()
- Share::getBackend()
- Share::getExpireInterval()
- Share::getItemShared()
- Share::getItemSharedWithBySource()
- Share::getItemSharedWithUser()
- Share::getItemsShared()
- Share::getItemsSharedWithUser()
- Share::getShareByToken()
- Share::getSharedItemsOwners()
- Share::getUsersSharingFile()
- Share::isEnabled()
- Share::isResharingAllowed()
- Share::registerBackend();
- Share::removeProtocolFromUrl()
- Share::resolveReShare()
- Share::setSendMailStatus();
- Share::shareItem()
- Share::shareItem()
- Share::shareWithGroupMembersOnly()

could be adressed in a separate PR

**Comment by PVince81** [comment]
rebased

**Comment by PVince81** [comment]
- [ ] check and remove `sendRemoteShare`

**Comment by PVince81** [comment]
@ownclouders rebase

**Comment by ownclouders** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by PVince81** [comment]
so we get more of the new acceptance tests to see if this PR breaks more ^

**Comment by ownclouders** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

**Comment by PVince81** [comment]
Rebased and resolved conflicts

**Comment by PVince81** [comment]
- [ ] fix ShareesController::isRemoteSharingAllowed: need to move away from `getBackend()` as we killed the share backend thing (we only have share providers now)

**Comment by PVince81** [comment]
Ok, so this PR is likely already breaking share email feature.

- [ ] Need to fix guest app and MailNotifications to stop using getItemSharedWithUser()

**Comment by PVince81** [comment]
- [ ] fix classifier to use share 2.0: https://github.com/owncloud/files_classifier/issues/24

**Comment by PVince81** [comment]
I'll now preemptively remove some methods, the apps that will break are listed as checkboxes.

**Comment by PVince81** [comment]
Might be possible to get rid of "informRecipient" completely if we don't want it in OC 11... Or rewire it to send regular notifications.

**Comment by PVince81** [comment]
the remaining difficult topics:
- getShareByToken
- getUsersSharingFile which we'll need to somehow move to the ShareManager. A lot depend on this.

**Comment by PVince81** [comment]
- [ ] remove obsolete unit tests

**Comment by PVince81** [comment]
clarification: we cannot merge this without breaking core stuff, need to solve the mail notifications part: https://github.com/owncloud/core/issues/32494

**Comment by PVince81** [comment]
removed many obsolete tests from ShareTest.php and fixed the other failures, and rebased

**Comment by PVince81** [comment]
5200 lines removed in total :tada: 

tests will still fail because of MailNotifications using a removed API: https://github.com/owncloud/core/pull/26608#issuecomment-428635185

there are acceptance tests related to setting share settings that fail, need to investigate if related to this PR or already on master

once this is done I'd say we move on and merge

**Comment by PVince81** [comment]
Rebased since https://github.com/owncloud/core/issues/32494 was merged. Expecting all tests to be green now.

**Comment by PVince81** [comment]
seems I forgot to grep in "settings" when killing APIs:
```
{"reqId":"teYh83ZViCZbzxzQ9rlo","level":3,"time":"2018-10-18T12:36:58+00:00","remoteAddr":"127.0.0.1","user":"admin","app":"index","method":"GET","url":"\/owncloud\/index.php\/settings\/admin?sectionid=sharing","message":"Exception: {\"Exception\":\"Error\",\"Message\":\"Call to undefined method OC\\\\Share\\\\Share::shareWithGroupMembersOnly()\",\"Code\":0,\"Trace\":\"#0 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/settings\\\/Panels\\\/Admin\\\/FileSharing.php(96): OC\\\\Settings\\\\Panels\\\\Helper->shareWithGroupMembersOnly()\\n#1 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/settings\\\/Controller\\\/SettingsPageController.php(167): OC\\\\Settings\\\\Panels\\\\Admin\\\\FileSharing->getPanel()\\n#2 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/settings\\\/Controller\\\/SettingsPageController.php(104): OC\\\\Settings\\\\Controller\\\\SettingsPageController->getPanelsData(Array)\\n#3 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(153): OC\\\\Settings\\\\Controller\\\\SettingsPageController->getAdmin('sharing')\\n#4 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(85): OC\\\\AppFramework\\\\Http\\\\Dispatcher->executeController(Object(OC\\\\Settings\\\\Controller\\\\SettingsPageController), 'getAdmin')\\n#5 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/App.php(100): OC\\\\AppFramework\\\\Http\\\\Dispatcher->dispatch(Object(OC\\\\Settings\\\\Controller\\\\SettingsPageController), 'getAdmin')\\n#6 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Routing\\\/RouteActionHandler.php(47): OC\\\\AppFramework\\\\App::main('SettingsPageCon...', 'getAdmin', Object(OC\\\\AppFramework\\\\DependencyInjection\\\\DIContainer), Array)\\n#7 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/Route\\\/Router.php(342): OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler->__invoke(Array)\\n#8 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/base.php(907): OC\\\\Route\\\\Router->match('\\\/settings\\\/admin')\\n#9 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/index.php(54): OC::handleRequest()\\n#10 {main}\",\"File\":\"\\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/settings\\\/Panels\\\/Helper.php\",\"Line\":38}"}
{"reqId":"MaWKdI7n8MhzZCM75Ijw","level":0,"time":"2018-10-18T12:36:59+00:00","remoteAddr":"127.0.0.1","user":"admin","app":"cron","method":"GET","url":"\/owncloud\/index.php\/cron","message":"Started background job of class : OCA\\Files\\BackgroundJob\\CleanupPersistentFileLocks with arguments : "}
```

**Comment by PVince81** [comment]
all green :tada: 

I'd appreciate some review before merging @DeepDiver1975 @jvillafanez @VicDeo 

see "Description" in the original post and also the list of spaghettis I managed to untangle: https://github.com/owncloud/core/pull/26608#issuecomment-368016322

there are still some spaghettis to untangle which must be adressed separately as they are more involved

**Review by DeepDiver1975 — APPROVED** [review]

**Comment by PVince81** [comment]
raised https://github.com/owncloud/core/issues/33219 for part 2 with remaining items

**Comment by VicDeo** [comment]
@PVince81 please note that `\OCP\Share`, `\OCP\Share_Backend`, `\OCP\Share_Backend_Collection`, `\OCP\Share_Backend_File_Dependent` should be marked as deprecated in stable10

**Comment by PVince81** [comment]
@VicDeo thanks for the reminder, I'll submit a PR

**Comment by PVince81** [comment]
deprecation PR for stable10: https://github.com/owncloud/core/pull/33220

**Pull request merged**